docs/best_practices/naming.rst

```diff
@@ -2,52 +2,70 @@ Naming
+data writers to provide human-readable information about the contents of the ``neurodata_type``.
+
+It is recommended to use :wikipedia:`CamelCase <Camel_case>` when naming instances of schematic classes.
+
 
-This allows multiple objects of the same type to be stored side-by-side and allows data writers to provide
-human-readable information about the contents of the neurodata_type. If appropriate, simply use the name of the
-Neurodata Type as the name of that object.
 
-For instance, if you are placing an ElectricalSeries object in /acquisition that holds voltage traces for a
-multichannel recording, consider simply naming that object "ElectricalSeries". This is the default_name for that
-object, and naming it like this will increase your chances that analysis and visualization tools will operate
-seamlessly with you data.There may be cases where you have multiple neurodata instances of the same type in the same
-Group. In this case the instances must have unique names. If they are both equally important data sources, build upon
-the class name (e.g. "ElectricalSeries_1" and "ElectricalSeries_2"). If one of the instances is an extra of less
-importance, name that one something different (e.g. "ElectricalSeries" and "ElectricalSeries_extra_electrode").
+.. _best_practice_description:
 
-Names are not for storing meta-data. If you need to place other data of the same neurodata_type, you will need to
-choose another name. Keep in mind that meta-data should not be stored solely in the name of objects. It is OK to name
-an object something like “ElectricalSeries_large_array” however the name alone is not sufficient documentation. In this
-case, the source of the signal will be clear from the device of the rows from the linked electrodes table region, and
-you should also include any important distinguishing information in the description field of the object. Make an effort
-to make meta-data as explicit as possible. Good names help users but do not help applications parse your file.
+Metadata Descriptions
+~~~~~~~~~~~~~~~~~~~~~
+
+The ``name`` of an object is for identifying that object within the file; it is not for storing metadata. Instead, the
+``description`` field for most objects can be used to include any important distinguishing information. When writing
+metadata, make every effort to be as explicit and self-contained as possible in the explaining relevant details.
+
+It is OK to name an object something like ``ElectricalSeriesDuringSomeEvent``, however the name alone is not sufficient
+documentation of the condition ``DuringSomeEvent``. In this case, the source of the signal will be clear from the
+:nwb-schema:ref:`device` linkage in the rows of the passed :nwb-schema:ref:`sec-sec-dynamictableregion` subsetting
+the full :nwb-schema:ref:`ElectrodeTable <groups-general-extracellular-ephys-electrodes>`, so you would not need to
+add any explicit metadata explaining these details.
```

Comment:
What about the event? How should that be encoded?

Reply:
Added details about encoding the event